### PR TITLE
M3-2499 Change: Validate Rebuild from Image form before modal opens

### PR DESCRIPTION
**Pull request by johnwcallahan**
## Description

Previously in Linode Rebuild, the form wasn't being validated until the request was sent. This was because we were relying on the services library to do the validation. Because we have a confirmation modal as an intermediate step, the UX was weird.

I added Formik to RebuildFromImage so that the form is validate on the first "Rebuild" button press.

(If there is additional API validation that happens, it won't happen until the request is actually sent.)

**NOTE:** I upgraded `react-testing-library` to get around console errors. More details in the commit messages.

_Same change in Rebuild from StackScript coming in a separate PR._

## Type of Change
- Non breaking change ('update', 'change')

## Note to Reviewers

Test that Rebuilding a Linode from an Image works as expected.

### CODE REVIEW

File: src/features/linodes/LinodesDetail/LinodeRebuild/RebuildFromImage.tsx [hunk 1]
```diff
@@ -69,108 +86,136 @@ export const RebuildFromImage: React.StatelessComponent<
...
-      <RebuildDialog
-        isOpen={isDialogOpen}
-        isLoading={isLoading}
-        handleClose={() => setIsDialogOpen(false)}
-        handleSubmit={handleSubmit}
-      />
-    </Grid>
+    <Formik
+      initialValues={initialValues}
+      validationSchema={RebuildLinodeSchema}
```
Comment (Jskobos) [inline src/features/linodes/LinodesDetail/LinodeRebuild/RebuildFromImage.tsx]: I know client-side password validation is tricky, but I think we could at least add min and max to this schema.

Comment (johnwcallahan) [reply src/features/linodes/LinodesDetail/LinodeRebuild/RebuildFromImage.tsx]: Good call @Jskobos – I added min, max, and the character constraints.

File: src/services/linodes/linode.schema.ts [hunk 2]
```diff
@@ -22,6 +22,15 @@ const stackscript_data = array()
 //     otherwise: string().notRequired()
 //   });
 
+// Covers length and character requirements. Chain with other constraints using .concat().
+const rootPasswordValidation = string()
+  .min(6, 'Password must be between 6 and 128 characters.')
+  .max(128, 'Password must be between 6 and 128 characters.')
+  .matches(
+    /^(((?=.*[a-z])(?=.*[A-Z]))|((?=.*[a-z])(?=.*[0-9]))|((?=.*[A-Z])(?=.*[0-9]))|((?=.*[a-z])(?=.*[!"#$%&'()*+,-.\/:;<=>?@\[\]^_`{|}~\\]))|((?=.*[A-Z])(?=.*[!"#$%&'()*+,-.\/:;<=>?@\[\]^_`{|}~\\]))|((?=.*[0-9])(?=.*[!"#$%&'()*+,-.\/:;<=>?@\[\]^_`{|}~\\])))/,
```
Comment (Jskobos) [inline src/services/linodes/linode.schema.ts]: This scares me a little, but afaict it's working as it should

Comment (martinmckenna) [reply src/services/linodes/linode.schema.ts]: I agree. I think this should be left to the API validation

**Pull request merged**